src/streamlink/stream/dash_manifest.py

```diff
@@ -461,97 +592,127 @@ def segment_numbers(self):
-                                                              if self.root.suggestedPresentationDelay
-                                                              else 3))
-                publish_time = self.root.publishTime or EPOCH_START
-
-                # transform the time line in to a segment list
-                timeline = []
-                available_at = publish_time
-                for segment, n in reversed(list(zip(self.segmentTimeline.segments, count(self.startNumber)))):
-                    # the last segment in the timeline is the most recent
-                    # so, work backwards and calculate when each of the segments was
-                    # available, based on the durations relative to the publish time
-                    url = self.make_url(self.media(Time=segment.t, Number=n, **kwargs))
-                    duration = datetime.timedelta(seconds=segment.d / self.timescale)
-
-                    # once the suggested_delay is reach stop
-                    if self.root.timelines[self.parent.id] == -1 and publish_time - available_at >= suggested_delay:
-                        break
-
-                    timeline.append((url, available_at, segment.t))
-
-                    available_at -= duration  # walk backwards in time
-
-                # return the segments in chronological order
-                for url, available_at, t in reversed(timeline):
-                    if t > self.root.timelines[self.parent.id]:
-                        self.root.timelines[self.parent.id] = t
-                        yield (url, available_at)
+        if not self.segmentTimeline:
+            for number, available_at in self.segment_numbers():
+                url = self.make_url(self.media(Number=number, **kwargs))
+                yield url, available_at
+            return
 
-            else:
-                for segment, n in zip(self.segmentTimeline.segments, count(self.startNumber)):
-                    yield (self.make_url(self.media(Time=segment.t, Number=n, **kwargs)),
-                           datetime.datetime.now(tz=UTC))
+        if self.parent.id is None:
+            # workaround for invalid `self.root.timelines[self.parent.id]`
+            # creates a timeline for every mimeType instead of one for both
+            self.parent.id = self.parent.mimeType
 
-        else:
-            for number, available_at in self.segment_numbers():
-                yield (self.make_url(self.media(Number=number, **kwargs)),
-                       available_at)
+        log.debug(f"Generating segment timeline for {self.root.type} playlist (id={self.parent.id}))")
+
+        if self.root.type == "static":
+            for segment, n in zip(self.segmentTimeline.segments, count(self.startNumber)):
+                url = self.make_url(self.media(Time=segment.t, Number=n, **kwargs))
+                available_at = datetime.datetime.now(tz=UTC)  # TODO: replace with EPOCH_START ?!
```

Comment:
I wasn't 100% sure here, and since I didn't want to include any logic changes, I added the TODO comment.

Manifests with static timelines should have segments always available, so I don't know why the current timestamp (even iterated in the loop) is being yielded here instead of just `EPOCH_START`.